web/app/scripts/workspace/workspace-wrapper.js

```diff
@@ -116,7 +116,7 @@ angular.module('biggraph')
           const dst = this.boxes[i];
           const inputs = dst.instance.inputs;
           for (let inputName in inputs) {
-            if (inputs.hasOwnProperty(inputName)) {
+            if (Object.prototype.hasOwnProperty.call(inputs, inputName)) {
```

Comment:
Why is this necessary? The old version certainly looked better... :)

Reply:
> Why is this necessary? The old version certainly looked better... :)

Yeah, this looks so bad. Read more on the [no-prototype-builtins](https://eslint.org/docs/rules/no-prototype-builtins) ESLint rule and the [Iterate through object properties](https://stackoverflow.com/questions/8312459/iterate-through-object-properties) Stack Overflow question.

I think the short story is that the object may have stuff we want to ignore from the Object prototype, so we need the check. But it also may _not_ have stuff from that prototype, in which case `inputs.hasOwnProperty` would be undefined.

Now that I look more closely, it looks like iterating over [`Object.entries(inputs)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries) would look much nicer. I'll switch to that.